src/main/java/com/cag/cagbackendapi/services/user/impl/UserService.java

```diff
@@ -36,4 +40,27 @@ public UserResponseDto registerUser(RegisterUserRequestDto registerUserRequestDt
 
         return userDao.saveUser(registerUserRequestDto);
     }
+
+    public UserResponseDto getByUserId(String userId) {
+
+        if(userId == null || userId == "") {
+            throw new BadRequestException("Invalid userId", null);
+        }
+
+        UUID userUUID;
+
+        try {
+            userUUID = UUID.fromString(userId);
+        } catch(Exception ex){
+            throw new BadRequestException("Invalid userId", null);
+        }
+
+        var userResponseDto = userDao.getUser(userUUID);
+
+        if(userResponseDto == null) {
+            throw new NotFoundException("User not found", null);
```

Comment:
You can put the "User not found." message into the constants file in the DetailedErrorMessages object and reference it here as:DetailedErrorMessages.USER_NOT_FOUND

Reply:
Please put the "User not found" error message in the constants file!